hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/replicas/chunk/ChunkKeyHandler.java

```diff
@@ -95,113 +90,155 @@ protected void execute(OzoneClient client, OzoneAddress address)
+                      byte[] slice = Arrays.copyOfRange(stripeBytes, i, i + checksumLen);
+                      jsonGen.writeString(StringUtils.byteToHexString(slice));
+                    }
+                    jsonGen.writeEndArray();
+                  }
+                  jsonGen.writeEndObject();
+                }
 
-                ArrayNode stripeChecksums = chunkNode.putArray("stripeChecksum");
-                for (int i = 0; i <= stripeBytes.length - checksumLen; i += checksumLen) {
-                  byte[] slice = Arrays.copyOfRange(stripeBytes, i, i + checksumLen);
-                  stripeChecksums.add(StringUtils.byteToHexString(slice));
+                jsonGen.writeEndArray(); // End chunks array
+                jsonGen.writeEndObject(); // End blockData object
+
+                if (isECKey) {
+                  int replicaIndex = keyPipeline.getReplicaIndex(datanodeDetails);
+                  int dataCount = ((ECReplicationConfig) keyPipeline.getReplicationConfig()).getData();
+                  // Index is 1-based,
+                  // e.g. for RS-3-2 we will have data indexes 1,2,3 and parity indexes 4,5
+                  ChunkType chunkType = (replicaIndex > dataCount) ? ChunkType.PARITY : ChunkType.DATA;
+                  jsonGen.writeStringField("chunkType", chunkType.name());
+                  jsonGen.writeNumberField("replicaIndex", replicaIndex);
                 }
-              }
-            }
+                jsonGen.writeEndObject(); // End this datanode's response object
 
-            if (isECKey) {
-              int replicaIndex = keyPipeline.getReplicaIndex(entry.getKey());
-              int dataCount = ((ECReplicationConfig) keyPipeline.getReplicationConfig()).getData();
-              // Index is 1-based,
-              // e.g. for RS-3-2 we will have data indexes 1,2,3 and parity indexes 4,5
-              ChunkType chunkType = (replicaIndex > dataCount) ? ChunkType.PARITY : ChunkType.DATA;
-              jsonObj.put("chunkType", chunkType.name());
-              jsonObj.put("replicaIndex", replicaIndex);
+                jsonGen.flush();
+              } catch (Exception e) {
+                System.err.printf("Error getting block from datanode %s: %s%n",
+                    datanodeDetails.getHostName(), e.getMessage());
+              }
             }
+          } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+          } finally {
+            xceiverClientManager.releaseClientForReadData(xceiverClient, false);
           }
-        } catch (InterruptedException e) {
-          throw new RuntimeException(e);
-        } finally {
-          xceiverClientManager.releaseClientForReadData(xceiverClient, false);
+
+          jsonGen.writeEndArray();
         }
+
+        jsonGen.writeEndArray(); // End keyLocations array
+        jsonGen.writeEndObject(); // End root object
+        jsonGen.flush();
```

Comment:
We should print a newline here so the shell prompt does not end up on the same line as the closing json bracket. Putting it inside the try-with-resources should work. Once this block ends jackson will close stdout and we won't be able to print anything. Double check this in your shell though.
```suggestion
        jsonGen.flush();
        System.out.println();
```